src/main/kotlin/fr/yvernault/rexa/service/XnatService.kt

```
@@ -0,0 +1,115 @@
+package fr.yvernault.rexa.service
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.github.kittinunf.fuel.core.FuelManager
+import com.github.kittinunf.fuel.core.Headers
+import com.github.kittinunf.fuel.core.extensions.authentication
+import com.github.kittinunf.fuel.httpGet
+import com.github.kittinunf.result.Result
+import fr.yvernault.rexa.XnatGlossary
+import fr.yvernault.rexa.model.Assessor
+import fr.yvernault.rexa.model.Scan
+import fr.yvernault.rexa.model.XnatProperties
+import org.slf4j.LoggerFactory
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.http.MediaType
+import org.springframework.stereotype.Service
+
+@Service
+@EnableConfigurationProperties(XnatProperties::class)
+class XnatService(private val properties: XnatProperties) {
+
+    private val logger = LoggerFactory.getLogger(XnatService::class.java)
+
+    init {
+        FuelManager.instance.basePath = properties.url
+    }
+
+    fun getScansForProject(projectId: String): List<Scan> {
+
+        val user = properties.user
+        val password = properties.password
+
+        var sessionList: List<Scan> = emptyList()
+
+        val httpAsync = "${XnatGlossary.urlArchiveLabel}?project=$projectId${XnatGlossary.scanUrlLabel}"
+                .httpGet()
+                .header(Headers.ACCEPT to MediaType.APPLICATION_JSON)
+                .authentication()
+                .basic(user, password)
+                .responseString() { request, response, result ->
+                    when (result) {
+                        is Result.Failure -> {
+                            val ex = result.getException()
+                            logger.error(ex.message)
```

We should throw an exception here if it's fail.